### PR TITLE
Add react-native-safe-area

### PR DESCRIPTION
**Pull request by pvinis**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by pvinis** [comment]
Travis gives
```
ERROR: 46:5  expect  TypeScript@next compile error: 
A parameter initializer is only allowed in a function or constructor implementation.
ERROR: 47:5  expect  TypeScript@next compile error: 
A parameter initializer is only allowed in a function or constructor implementation.
```
for
```

export function withSafeArea(
    WrappedComponent: React.ComponentType<any>,
    applyTo: 'margin' | 'padding' | 'absolutePosition' | 'contentInset' = 'margin',
    direction: Direction = 'all',
): React.ComponentType<any>;
```
on the lines of `applyTo` and `direction`.

What's the solution here? Is this a problem with my code?

**Comment by pvinis** [comment]
@uniqueiniquity

File: types/react-native-safe-area/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    | 'horizontal'
+    | 'verticalAndLeft'
+    | 'verticalAndRight'
+    | 'horizontalAndTop'
+    | 'horizontalAndBottom'
+    | 'all';
+
+export function withSafeArea(
+    WrappedComponent: React.ComponentType<any>,
+    applyTo: 'margin' | 'padding' | 'absolutePosition' | 'contentInset' = 'margin',
```
Comment (uniqueiniquity) [inline types/react-native-safe-area/index.d.ts]: The `= 'margin'` part of this is not allowed in a .d.ts file.
You should instead use `applyTo?: 'margin' | 'padding' | 'absolutePosition' | 'contentInset'`
You should be able to use JSDoc comments to document the default value that is provided if you'd like.

**Review by uniqueiniquity — CHANGES_REQUESTED** [review]

**Comment by pvinis** [comment]
ah so the typings don't show the default. alright. thanks.

**Comment by pvinis** [comment]
done.

File: types/react-native-safe-area/index.d.ts (hunk 1 above)
Comment (pvinis) [reply types/react-native-safe-area/index.d.ts]: done

**Comment by pvinis** [comment]
@uniqueiniquity your suggestions helped. thank you. you can approve if it's required.

**Review by uniqueiniquity — APPROVED** [review]

**Pull request merged**